src/components/budget/budget.service.ts

```diff
@@ -356,6 +356,39 @@ export class BudgetService {
   async update(input: UpdateBudget, session: ISession): Promise<Budget> {
     const budget = await this.readOne(input.id, session);
 
+    //574 - Budget records are only editable if the budget is pending
+    if (budget.status.includes(BudgetStatus.Current)) {
+      throw new BadRequestException('budget can not be modified');
+    }
+
+    //Get Project.Status
+    const projectStatusQuery = this.db
+      .query()
+      .match(matchSession(session, { withAclRead: 'canReadBudgets' }))
+      .match([
+        node('budget', 'Budget', { active: true, id: input.id }),
+        relation('in', '', 'budget', {
+          active: true,
+        }),
+        node('project', 'Project', { active: true }),
+        relation('out', '', 'status', { active: true }),
+        node('status', 'Property', { active: true }),
+      ]);
+    projectStatusQuery.return([
+      {
+        project: [{ id: 'id' }],
+        status: [{ value: 'status' }],
+      },
+    ]);
+
+    const readProject = await projectStatusQuery.first();
+    //Budget records are only editable if Project.status not active - ProjectStatus.Active
+    if (readProject?.status.includes('Active')) {
+      throw new BadRequestException(
+        'budget of active project can not be modified '
+      );
+    }
```

Comment:
A main point of #574 is to separate budget changes from project status.
Since `updateBudget` only updates the status and we'll be doing that via workflow, I think this API can just go away. And that just leaves this method as a helper for internal use.

Where budgets really change is on their records. `updateRecord` is what needs to check for whether the budget status is _Pending_.

Reply:
ok, I will change that.

Reply:
done the changes